### DISCUSSION TITLE
Set refresh rate to specific one instead of max refresh rate

### DISCUSSION THREAD

**Discussion by People-11** [seerge/g-helper#1067] [Q&A] [answered]
The thing is, I set a 120Hz refresh rate within the NVIDIA control panel and use it for some specific reason (my screen tops out at 165Hz). But when I switch the refresh rate within G-Helper, G-Helper always sets my screen to 165 Hz. so I'd like to ask the question, is it possible for G-Helper to switch to a specified refresh rate (provided it's present in the system settings) instead of always switching to the highest refresh rate? If not, and I wish to directly modify the G-Helper's program to fix the maximum refresh rate to 120Hz, where do I change the code?

**Comment by seerge** [comment]
Just don't select Auto in G-Helper (select either 60hz or max rate) and then set your 120Hz in control panel. App won't be overriding it. It overrides refresh rate on start only for Auto.

**Reply by People-11** [reply]
Can I fix the max refresh rate to 120 by editing the code?

**Reply by seerge** [reply]
You can, but currently app detects max on the fly :) so it's not hardcoded anyhow

**Reply by People-11** [reply]
I would appreciate it if you could provide me with the location of the code that needs to be changed, I am not good at reading C# code.

**Reply by People-11** [reply]
maybe its in https://github.com/seerge/g-helper/blob/6d6a9c68f2835a7c7d85becd5049e0e2e9a31fe5/app/Display/ScreenControl.cs#L40 , lemme try. Plz correct me if I am wrong :).

Its working, but need to do more around

**Reply by seerge** [reply, marked as answer]
It's 
https://github.com/seerge/g-helper/blob/6d6a9c68f2835a7c7d85becd5049e0e2e9a31fe5/app/Display/ScreenControl.cs#L35C19-L35C19

replace with

```
frequency   = 120;
```

**Comment by People-11** [comment]
The reasons I set it to 120Hz:

1. this machine doesn't have G-Sync and it's graphics card isn't strong enough to support 2K 165FPS gaming, so once it fluctuates in frame rate during a game, it will visually lag badly. So 2K120FPS is a good option that doesn't drop too many frames while remaining crisp and smooth.

2. If you do 120FPS or 60FPS screen recording on 165FPS content, your video will have 'judder', which simply means uneven frame skipping because the 165FPS screen cannot be evenly distributed to 120FPS or 60FPS. You can notice that Windows Screen Recording actually records 55FPS video to avoid this problem.

**Comment by vwgeiser** [comment]
@People-11 Where was this code located on your machine? I only see `config.json` `config.json` `log.txt` in %AppData%/Roaming/GHelper? my G14 only has a max of 120hz and it is causing a jittering black box to appear temporarily on startup and plugged in (probably becuase the refresh rate is set too high?) The problem is halted by setting the max refresh to 60hz temporarily. 

Ghelper seems to think that my machine can run at 144hz?
![image](https://github.com/user-attachments/assets/8aab1688-8773-49c9-91e0-a7dbbbe001a4)

**Reply by seerge** [reply]
@vwgeiser hello, there is no need to modify config in your case. Just set 60hz in the app and it won’t change:)

**Reply by vwgeiser** [reply]
@seerge Ah I see, this is my second G14, today I learned that it has a 144hz refresh rate as compared to my first one which had 120hz...
